### ISSUE TITLE
Some DocuSign API calls don't like null as object property value

### ISSUE THREAD

**Issue by mmallis87** [docusign/docusign-node-client#47]
The Node SDK sets up some object properties to null (Such as EnvelopeDefinition), which makes the API call fail with 404 ("The request body is missing or improperly formatted. Null object cannot be converted to a value type.").

A workaround is to loop through the object properties recursively and delete the properties that has null values in them, before calling DocuSign API. Here is a code snippet:

```
removeNulls(envelopeDefinition);
envelopesApi.createEnvelope(dsAccountId, envelopeDefinition, null, function(error, envelopeSummary, response) {
}
```

Here is the code for removeNulls helper function:

```
removeNulls = function(obj) {
  var isArray = obj instanceof Array;
  for (var k in obj) {
    if (obj[k] === null) isArray ? obj.splice(k, 1) : delete obj[k];
    else if (typeof obj[k] == "object") removeNulls(obj[k]);
    if (isArray && obj.length == k) removeNulls(obj);
  }
  return obj;
}
```

**Comment by NullVoxPopuli** [comment]
any updates on this?

**Comment by mmallis87** [comment]
Unfortunately nothing yet. The issue has been tracked to our internal bug tracker, but didn't make its way yet because of prioritization. Especially that there is a workaround (as ugly as it is).

**Comment by motevets** [comment]
Any update on this?

**Comment by mmallis87** [comment]
@NullVoxPopuli @motevets the SDK bugs were prioritized and this is one is being fixed right now. A minor release should be deployed in the coming days and I will let you know once it's out.

**Comment by mmallis87** [comment]
Shipped in v2.1.0.
